queue/abstract/driver/fifottl.lua

```diff
@@ -300,12 +300,18 @@ function method.release(self, id, opts)
     if task == nil then
         return
     end
+    if opts.touch_ttl == nil then opts.touch_ttl = true end
     if opts.delay ~= nil and opts.delay > 0 then
-        task = self.space:update(id, {
+        local upd = {
             { '=', i_status, state.DELAYED },
-            { '=', i_next_event, util.event_time(opts.delay) },
+            { '=', i_next_event, util.event_time(opts.delay) }
+        }
+        if opts.touch_ttl then
+            table.insert(upd,
             { '+', i_ttl, util.time(opts.delay) }
-        })
+            )
+        end
+        task = self.space:update(id, upd)
```

Comment:
I would name a new option in a way to make its default value `false`: say, `keep_ttl`. It looks more natural for me.

I guess the code hunk may be written in a simpler way (didn't tested):

```lua
task = self.space:update(id, {
    { '=', i_status, state.DELAYED },
    { '=', i_next_event, util.event_time(opts.delay) },
    not opts.keep_ttl and { '+', i_ttl, util.time(opts.delay) } or nil,
})
```

Reply:
Thanks for the fine advise. Renamed